src/device-state.ts

```diff
@@ -128,14 +128,64 @@ function createDeviceStateRouter() {
 			),
 	);
 
-	router.patch('/v1/device/host-config', (req, res) =>
-		hostConfig
-			.patch(req.body)
-			.then(() => res.status(200).send('OK'))
-			.catch((err) =>
-				res.status(503).send(err?.message ?? err ?? 'Unknown error'),
-			),
-	);
+	router.patch('/v1/device/host-config', async (req, res) => {
+		// Validate user input, returning 400 with malformed PATCH body
+		try {
+			if (!req.body.network) {
+				throw new Error("Key 'network' must exist in PATCH body");
+			}
+
+			const { hostname, proxy } = req.body.network;
+
+			// If hostname is an empty string, return first 7 digits of device uuid
+			if (hostname === '') {
+				const uuid = await config.get('uuid');
+				req.body.network.hostname = uuid?.slice(0, 7);
+			}
+
+			// Validate proxy fields, if they exist
+			if (proxy && Object.keys(proxy).length) {
+				// TODO: Should probably support changing local_port to any value
```

Comment:
Restricting the allowed configurations might break things for users @cywang117, what is your reasoning behind this? I found at least one instance where a user was using `local_ip` https://forums.balena.io/t/redsocks-proxy-not-working/220377

Unless there is a good reason I would allow everything redsocks allows. If we know redsocks doesn't crash with unknown fields we might not bother checking at all, since this is something we will need to keep maintaining if the OS redsocks version changes.

Reply:
If we end up removing the validation, you can disregard my previous comment about moving the checks to `host-config`

Reply:
@pipex I was simply building on top of legacy code, [which sets `local_ip` and `local_port` as static fields](https://github.com/balena-io/balena-supervisor/blob/2fa8156b31c60d56d07bee28bd6ff5fab8cfece2/src/host-config.ts#L19). 

However I do agree that we should support all fields being changed. I figured it would make for a separate PR, but forgot to mention the topic. Since we are implementing feature parity on V2 from V1, this issue would be relevant up to and possible after V3 comes out. I've raised an issue [here - TODO]().

I haven't researched into redsocks deeply enough to know 100%, but AFAIK unknown fields would be ignored. There are also a lot more config options for redsocks that are not supported here, even if we allow unknown fields. The top one that comes to mind is [support for `redudp`](https://github.com/darkk/redsocks/blob/master/redsocks.conf.example#L117), which would be a separate bracket in the `.conf` file. Should we enable support for this sort of thing as well?

I do have more context-gathering questions for redsocks-related things, but I'll send them over in the FD supervisor thread.

Reply:
You're right, I hadn't seen that part of the code before. Maybe blacklist proxy fields instead of whitelisting is the option?

> I was simply building on top of legacy code, which sets local_ip and local_port as static fields.

No, let's keep it as is for now, for that sort of change is better to go through a product discussion and for that is better to have user evidence, otherwise we end up with extra code we will forced to maintain.
> Should we enable support for this sort of thing as well?

Reply:
> Maybe blacklist proxy fields instead of whitelisting is the option?

Sorry, could you clarify? I thought 

```typescript
if (
	Object.keys(proxy).filter(
		(key) => !constants.validV1PatchProxyFields.includes(key),
	).length > 0
) {...}
```

functioned as blacklisting? ([device-state.ts line 151 in PR](https://github.com/balena-io/balena-supervisor/pull/1589/commits/84eee4381aea81f932c7417245a378c64a99e76f#diff-28bbeff4b860b9fa9095237759c4b8891ef72dd96acfed6c7241641c964439c4R150))

Unless you're talking about adding local_ip and local_port support as well?

Reply:
If I understand correctly, that code checks that the provided values match the values on a 'valid field list', so is more whitelisting. Blacklisting would be something like

```
if (
	Object.keys(proxy).filter(
		(key) => constants.disalllowedV1PatchProxyFields.includes(key),
	).length > 0
) {...}
```

Does this make more sense?

Reply:
Thanks, makes sense. I've updated the commit :)

Reply:
@pipex  I've also changed the route from throwing errors on invalid fields to logging in the supervisor console, per our discussion yesterday. Let me know if it works for you.